### PR TITLE
Logout user/agent when being disabled

### PR DESCRIPTION
**Pull request by lchanmann**
Forcefully logout user when being disabled:

- [x] Logout the user session
- [x] Logout all agents who currently login and act as the user

Extracted from: https://github.com/nanoPayinc/NANOPAY/pull/5354

### CODE REVIEW

File: src/foam/nanos/auth/LogoutDisabledUserDAO.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+  javaImports: [
+    'foam.dao.ArraySink',
+    'foam.dao.DAO',
+    'foam.mlang.MLang',
+    'foam.nanos.auth.AuthService',
+    'foam.nanos.auth.User',
+    'foam.nanos.session.Session',
+    'foam.util.SafetyUtil',
+    'net.nanopay.model.Business',
```
Comment (jlhughes) [inline src/foam/nanos/auth/LogoutDisabledUserDAO.js]: Please remove nanopay references

File: src/foam/nanos/auth/LogoutDisabledUserDAO.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      name: 'logoutAgent',
+      javaReturns: 'void',
+      args: [
+        { of: 'User', name: 'user' },
+        { of: 'DAO', name: 'entitiesDAO'}
+      ],
+      javaCode: `
+        ArraySink sink = (ArraySink) entitiesDAO.select(new ArraySink());
+        List<Business> businesses = sink.getArray();
+
```
Comment (jlhughes) [inline src/foam/nanos/auth/LogoutDisabledUserDAO.js:48]: This nanopay business logic will have to be added on a nanopay side decorator.

**Review by jlhughes — CHANGES_REQUESTED** [review]

File: src/foam/nanos/auth/LogoutDisabledUserDAO.js (hunk 1 above)
Comment (lchanmann) [reply src/foam/nanos/auth/LogoutDisabledUserDAO.js]: Fixed

File: src/foam/nanos/auth/LogoutDisabledUserDAO.js (hunk 2 above)
Comment (lchanmann) [reply src/foam/nanos/auth/LogoutDisabledUserDAO.js:48]: Fixed

**Comment by lchanmann** [comment]
Comments addressed. Please take another look, thanks @jlhughes

**Review by jlhughes — APPROVED** [review]

**Pull request merged**